### PR TITLE
🏃Add release binaries target to Makefile

### LINKED ISSUE #1362: `make release` needs to build clusterctl
/kind bug

**What steps did you take and what happened:**
1. Ran `make release`
1. It didn't build `clusterctl`

**What did you expect to happen:**
1. Build `clusterctl` for the supported ARCHes

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1362

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -187,19 +187,36 @@ set-manifest-image:
...
-	kustomize build config/default > out/cluster-api-components.yaml
+	$(MAKE) release-manifests
+	$(MAKE) release-binaries
+
+.PHONY: release-manifests
+release-manifests: $(RELEASE_DIR) ## Builds the manifests to publish with a release
+	kustomize build config/default > $(RELEASE_DIR)/cluster-api-components.yaml
+
+.PHONY: release-binaries
+release-binaries: $(addprefix release-binary-,$(RELEASE_BINARIES)) ## Builds the binaries to publish with a release
```
Comment (detiber) [inline Makefile]: Are we expecting multiple binaries here? otherwise these seems like overkill.

**Comment by ncdc** [comment]
Should we prevent `make release` from proceeding if the local checkout has any modified or extra files? Or if you don't have a tag checked out?

File: Makefile [hunk 2]
```diff
@@ -187,19 +187,36 @@ set-manifest-image:
...
+
+release-binary-%: $(RELEASE_DIR)
+	@for os in $(RELEASE_GOOS); do \
+		CGO_ENABLED=0 GOOS=$${os} GOARCH=amd64 go build -a -ldflags '-extldflags "-static"' -o $(RELEASE_DIR)/$*-$${os}-amd64; \
```
Comment (ncdc) [inline Makefile]: Should we build using docker so we're not at the mercy of your locally installed go?

File: Makefile (hunk 1 above)
Comment (vincepri) [reply Makefile]: It's gonna be super easy to port this to other repos in this way, that's why I went down this path

File: Makefile (hunk 2 above)
Comment (vincepri) [reply Makefile]: Do you have an example? Or can we do this later?

Comment (ncdc) [reply Makefile]: Most/all of this is pre go modules, but here's how we set up velero: https://github.com/heptio/velero/blob/29f29f9d614f396dbb310423b65ae16ae129f451/Makefile#L95-L130.

My preference is to only merge a PR that builds in docker. I'm happy to take this over if you want.

Comment (vincepri) [reply Makefile]: Added the build in docker command, ptal :)

File: Makefile [hunk 3]
```diff
@@ -187,19 +187,42 @@ set-manifest-image:
 ## --------------------------------------
 
 RELEASE_TAG := $(shell git describe --abbrev=0 2>/dev/null)
+RELEASE_DIR := out
+
+$(RELEASE_DIR):
```
Comment (ncdc) [inline Makefile]: Does this need to be phony?

Comment (vincepri) [reply Makefile]: How so? It'd only create it if it doesn't exist, unless I'm missing something

Comment (ncdc) [reply Makefile]: Nevermind

**Pull request merged**